write.go

```diff
@@ -5,16 +5,24 @@ import (
 	"crypto/sha256"
 	"encoding/json"
 	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
 	"github.com/Rhymen/go-whatsapp/binary"
 	"github.com/Rhymen/go-whatsapp/crypto/cbc"
 	"github.com/gorilla/websocket"
 	"github.com/pkg/errors"
-	"strconv"
-	"time"
 )
 
+var writerLock sync.RWMutex
+
 //writeJson enqueues a json message into the writeChan
 func (wac *Conn) writeJson(data []interface{}) (<-chan string, error) {
+
+	writerLock.Lock()
```

Comment:
The writerLock is a global lock, but the Conn is an instance.

The lock should be scoped to the instance only, don't you think?

Reply:
Sure. What do you think now?